packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -113,108 +112,108 @@ const withInfiniteScroll = (
         showSeeMore: false,
       };
 
-      this.onItemsFetch = props.onItemsFetch;
-      this.observer = new IntersectionObserver(this.handleIntersection, {
-        threshold: 1,
+      this.props.dataSource.onDataChange((updatedElements: Array<any>) => {
+        const { index } = this.state;
```

Comment:
To be honest I think if we don't move this to the dataSource is better to keep the version you had before (calling fetch again) as this will give more flexibility in case someone is implementing a dataSource that only gets notified of new data but doesn't get the actual data and needs to fetch it again